ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -250,20 +250,13 @@ class ActionTuple:
     Continuous and discrete actions are numpy arrays of type float32 and
     int32, respectively and are type checked on construction.
     Dimensions are of (n_agents, continuous_size) and (n_agents, discrete_size),
-    respectively.
+    respectively. Note, this also holds when continuous or discrete size is
+    zero.
     """
 
-    def __init__(
-        self,
-        continuous: Optional[np.ndarray] = None,
-        discrete: Optional[np.ndarray] = None,
-    ):
-        if continuous is not None and continuous.dtype != np.float32:
-            continuous = continuous.astype(np.float32, copy=False)
-        self._continuous = continuous
-        if discrete is not None and discrete.dtype != np.int32:
-            discrete = discrete.astype(np.int32, copy=False)
-        self._discrete = discrete
+    def __init__(self):
+        self._continuous = None
+        self._discrete = None
```

Comment:
How about keeping the continuous and discrete arguments, and adding e.g.
```python
if continuous is not None:
  self.add_continuous(continuous)
# TODO same for discrete
```
?

Reply:
+1